static/js/components/VideoPlayer.js

```diff
@@ -216,6 +225,34 @@ export default class VideoPlayer extends React.Component<*, void> {
     )
   }
 
+  imageLoaded = (loaded: boolean) => {
+    const { video } = this.props
+    if (!loaded && video.sources.length > 0) {
+      this.player
+        .reset()
+        .src(video.sources)
+        .fluid(false)
+    }
+  }
+
+  imageExists(url: string, callback: Function) {
+    const img = new Image()
+    img.onload = function() {
+      callback(true)
+    }
+    img.onerror = function() {
+      callback(false)
+    }
+    img.src = url
+  }
```

Comment:
seems to me like there's no reason to call `imageLoaded` with `true` as an argument - you could refactor the `imageLoaded` function to just do the `this.player.reset()...` business - then (I think) you might be able to get away with just doing:

```js
imageExists = (url: string) => {
  const img = new Image()
  img.onerror = this.imageLoaded
  img.src = url
}
```

Reply:
✅